solution/0300-0399/0323.Number of Connected Components in an Undirected Graph/README_EN.md

```diff
@@ -487,4 +487,47 @@ function countComponents(n: number, edges: number[][]): number {
 
 <!-- solution:end -->
 
+<!-- solution:start -->
+
+### Solution 3: BFS
+
+<!-- tabs:start -->
+
+#### TypeScript
+
+```ts
+function countComponents(n: number, edges: number[][]): number {
+    const g: number[][] = Array.from({ length: n }, () => []);
+    const vis = new Set<number>();
+    let ans = 0;
+
+    for (const [i, j] of edges) {
+        g[i].push(j);
+        g[j].push(i);
+    }
+
+    const dfs = (i: number) => {
+        if (vis.has(i)) return;
+
+        vis.add(i);
+        for (const j of g[i]) {
+            dfs(j);
+        }
+    };
+
+    for (let i = 0; i < n; i++) {
+        if (vis.has(+i)) continue;
+
+        dfs(+i);
```

Comment:
`i` is a number, so you don't need to use + to convert it.
```suggestion
        if (vis.has(i)) continue;

        dfs(i);
```

Reply:
Yes, just forgot to remove